app/data/data-travel-certificate/data-travel-certificate-public/src/main/kotlin/com/hedvig/android/data/travelcertificate/GetEligibleContractsWithAddressUseCase.kt

```diff
@@ -0,0 +1,43 @@
+package com.hedvig.android.data.travelcertificate
+
+import arrow.core.Either
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.cache.normalized.FetchPolicy
+import com.apollographql.apollo3.cache.normalized.fetchPolicy
+import com.hedvig.android.apollo.safeExecute
+import com.hedvig.android.apollo.toEither
+import com.hedvig.android.core.common.ErrorMessage
+import octopus.EligibleContractsWithAddressQuery
+
+interface GetEligibleContractsWithAddressUseCase {
+  suspend fun invoke(): Either<ErrorMessage, List<ContractEligibleWithAddress>>
+}
+
+internal class GetEligibleContractsWithAddressUseCaseImpl(
+  private val apolloClient: ApolloClient,
+) : GetEligibleContractsWithAddressUseCase {
+  override suspend fun invoke(): Either<ErrorMessage, List<ContractEligibleWithAddress>> {
+    return apolloClient.query(EligibleContractsWithAddressQuery())
+      .fetchPolicy(FetchPolicy.NetworkOnly)
+      .safeExecute()
+      .toEither(::ErrorMessage)
+      .map { data ->
+        data.currentMember.activeContracts.filter { contract ->
+          contract.supportsTravelCertificate
+        }.map { contract ->
+          val street =
+            data.currentMember.travelCertificateSpecifications.contractSpecifications.firstOrNull {
+              it.contractId == contract.id
+            }?.location?.street
+          val address = street ?: contract.exposureDisplayName.substringBefore("•")
+          // todo: add this check here bc location.street is nullable?
```

Comment:
Yeah that's a fair fallback here, do the right thing by default and fall back to the substring method in the odd case.

Since I expect this to never be hit unless there is an actual error on our backend, how about we log this as an error so that we make sure to address it if it ever happens?
```suggestion
          val address = street ?: contract.exposureDisplayName.substringBefore("•").also {
            logcat(LogPriority.ERROR) { "Received a travel certificate eligible contract, but the contractSpecifications did not include a valid street name" }
          }
          // todo: add this check here bc location.street is nullable?
```
Something like that? And if this actually gets hit in production there might be smth wrong with backend.